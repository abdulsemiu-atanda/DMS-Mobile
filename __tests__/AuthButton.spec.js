import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import AuthButton from '../components/shared/AuthButton.react'

describe('AuthButton', () => {
  describe.only('Login', () => {
    it('renders correctly when not logging in', () => {
      const tree = renderer.create(<AuthButton login />).toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree.children[0].children[0]).toEqual('Sign In')
    })

    it('renders spinner when logging in', () => {
      const tree = renderer.create(<AuthButton loggingIn login />).toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree.children[0].type).toEqual('ActivityIndicator')
    })
  })

  describe.only('SignUp', () => {
    it('renders correctly when not signing up', () => {
      const tree = renderer.create(<AuthButton />).toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree.children[0].children[0]).toEqual('Create')
    })

    it('renders spinner when signing up', () => {
      const tree = renderer.create(<AuthButton signingUp />).toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree.children[0].type).toEqual('ActivityIndicator')
    })
  })
})
