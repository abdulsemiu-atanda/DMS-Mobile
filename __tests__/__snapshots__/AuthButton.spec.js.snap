// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton Login renders correctly when not logging in 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <Text
    style={
      Object {
        "color": "#aacfff",
        "fontSize": 18,
      }
    }
  >
    Sign In
  </Text>
</View>
`;

exports[`AuthButton Login renders spinner when logging in 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <ActivityIndicator
    animating={true}
    color="#01f0b3"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`AuthButton SignUp renders correctly when not signing up 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <Text
    style={
      Object {
        "color": "#aacfff",
        "fontSize": 18,
      }
    }
  >
    Create
  </Text>
</View>
`;

exports[`AuthButton SignUp renders spinner when signing up 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <ActivityIndicator
    animating={true}
    color="#01f0b3"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
